src/network/remote.ts

```
@@ -238,23 +243,74 @@ export class HttpNetwork implements Network {
-    ).href;
-    this.timeout = config.timeout || 30 * util.SEC;
-
-    this.baseHeaders = {
-      "Content-Type": "application/json",
-      ...config.headers,
-    };
-    this.encoder = config.encoder ?? new JsonEncoder();
+    this.msgUrl = `${host}:${messageSourcePort}`;
+
+    this.group = group;
+    this.pid = pid;
+    this.timeout = timeout;
+    this.encoder = encoder;
+
+    this.headers = { "Content-Type": "application/json", ...headers };
+    if (auth) {
+      this.headers.Authorization = `Basic ${btoa(`${auth.username}:${auth.password}`)}`;
+    }
+
+    this.msgHeaders = {};
+    if (messageSourceAuth) {
+      this.msgHeaders.Authorization = `Basic ${btoa(`${messageSourceAuth.username}:${messageSourceAuth.password}`)}`;
+    }
+
+    this.eventSource = this.connect();
+  }
+
+  private connect() {
+    const url = new URL(`/${encodeURIComponent(this.group)}/${encodeURIComponent(this.pid)}`, this.msgUrl);
+    this.eventSource = new EventSource(url, {
+      fetch: (url, init) =>
+        fetch(url, {
+          ...init,
+          headers: {
+            ...init.headers,
+            ...this.msgHeaders,
+          },
+        }),
+    });
+
+    this.eventSource.addEventListener("message", (event) => {
+      let msg: Message;
+      const data = JSON.parse(event.data);
+
+      if ((data?.type === "invoke" || data?.type === "resume") && util.isTaskRecord(data?.task)) {
+        msg = { type: data.type, task: data.task };
+      } else if (data?.type === "notify" && util.isDurablePromiseRecord(data?.promise)) {
+        msg = { type: data.type, promise: this.mapPromiseDtoToRecord(data.promise) };
+      } else {
+        console.warn("Received invalid message:", data);
+        return;
+      }
+
+      this.recv(msg);
+    });
+
+    this.eventSource.addEventListener("error", () => {
+      console.log(`Networking. Cannot connect to [${this.msgUrl}]. Retrying in 5s.`);
+      setTimeout(() => this.connect(), 5000);
```

As part of this PR I needed to implement reconnect logic. This is really helpful, on error the connection was silently dropped.